block/manager.go

```diff
@@ -88,10 +89,11 @@ type Manager struct {
 // getInitialState tries to load lastState from Store, and if it's not available it reads GenesisDoc.
 func getInitialState(store store.Store, genesis *tmtypes.GenesisDoc) (types.State, error) {
 	s, err := store.LoadState()
-	if err != nil {
-		s, err = types.NewFromGenesisDoc(genesis)
+	if err == nil {
+		return s, nil
 	}
-	return s, err
+
+	return types.NewFromGenesisDoc(genesis)
```

Comment:
Seems like if we get an error not related to not having a genesis state, we'll still try to create a genesis state (Like happend to us before) without knowing the root cause of the error until debugging. 
My suggestion is to check for a specific error type if possible before creating `NewFromGensisDoc`